crates/uv/tests/it/lock_conflict.rs

```diff
@@ -15143,3 +15143,160 @@ fn avoids_exponential_lock_file_growth() -> Result<()> {
+    pyproject_toml.write_str(pyproject)?;
+
+    uv_snapshot!(context.filters(), context.lock(), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 4 packages in [TIME]
+    ");
+
+    let lock = context.read("uv.lock");
+    insta::with_settings!({
+        filters => context.filters(),
+    }, {
+        assert_snapshot!(
+            lock,
+            @r#"
+        version = 1
+        revision = 3
+        requires-python = "==3.12.*"
+        resolution-markers = [
+            "platform_machine != 'inapplicable' and extra != 'extra-5-debug-a' and extra == 'extra-5-debug-b'",
+            "platform_machine == 'inapplicable' and extra != 'extra-5-debug-a' and extra == 'extra-5-debug-b'",
+            "extra == 'extra-5-debug-a' and extra != 'extra-5-debug-b'",
+            "extra != 'extra-5-debug-a' and extra != 'extra-5-debug-b'",
+        ]
+        conflicts = [[
+            { package = "debug", extra = "a" },
+            { package = "debug", extra = "b" },
+        ]]
+
+        [options]
+        exclude-newer = "2025-02-06T00:00:00Z"
+
+        [[package]]
+        name = "debug"
+        version = "0.0.1"
+        source = { virtual = "." }
+
+        [package.optional-dependencies]
+        a = [
+            { name = "python-dateutil", version = "2.8.0", source = { registry = "https://pypi.org/simple" } },
+        ]
+        b = [
+            { name = "python-dateutil", version = "2.8.0", source = { registry = "https://pypi.org/simple" }, marker = "(platform_machine == 'inapplicable' and extra == 'extra-5-debug-b') or (extra == 'extra-5-debug-a' and extra == 'extra-5-debug-b')" },
+            { name = "python-dateutil", version = "2.8.1", source = { registry = "https://pypi.org/simple" }, marker = "(platform_machine != 'inapplicable' and extra == 'extra-5-debug-b') or (extra == 'extra-5-debug-a' and extra == 'extra-5-debug-b')" },
+        ]
+
+        [package.metadata]
+        requires-dist = [
+            { name = "python-dateutil", marker = "platform_machine == 'inapplicable' and extra == 'b'", specifier = "==2.8.0" },
+            { name = "python-dateutil", marker = "platform_machine != 'inapplicable' and extra == 'b'", specifier = "==2.8.1" },
+            { name = "python-dateutil", marker = "extra == 'a'", specifier = "==2.8.0" },
+        ]
+        provides-extras = ["a", "b"]
+
+        [[package]]
+        name = "python-dateutil"
+        version = "2.8.0"
+        source = { registry = "https://pypi.org/simple" }
+        resolution-markers = [
+            "platform_machine == 'inapplicable' and extra != 'extra-5-debug-a' and extra == 'extra-5-debug-b'",
+            "extra == 'extra-5-debug-a' and extra != 'extra-5-debug-b'",
+        ]
+        dependencies = [
+            { name = "six", marker = "(platform_machine == 'inapplicable' and extra == 'extra-5-debug-b') or extra == 'extra-5-debug-a'" },
```

Comment:
The marker is exactly the reachability of python-dateutil 2.8.0, we should be able to elide it.